app/controllers/api/v3/application_controller.rb

```diff
@@ -3,12 +3,22 @@ module V3
     class ApplicationController < ActionController::API
       rescue_from ActiveRecord::RecordNotFound, with: :jsonapi_404
 
+      before_action :check_disable_pagination
+
       def jsonapi_404
         render jsonapi: nil, status: :not_found
       end
 
     private
 
+      def check_disable_pagination
+        if params[:page] && params[:page][:per_page].to_i > Kaminari.config.default_per_page
+          unless respond_to?(:allowed_to_disable_pagination?) && allowed_to_disable_pagination?
+            params[:page][:per_page] = Kaminari.config.default_per_page
+          end
+        end
+      end
```

Comment:
How about using this as a guard:

```
return unless respond_to?(:allowed_to_disable_pagination?)
```

And then you can take it out of the unless on line 16, reducing it's Cognitive Complexity. :)

Reply:
Actually, could also have the guard:

```
return unless params[:page]
```

Reply:
i can do the second guard you suggested. however not the first guard as if that fails we need to fallback to the default

Reply:
This has now been changed. `ApplicationController` implements `allowed_to_disable_pagination ?` and returns false. Subclasses can override accordingly. The guards have now been put in place